src/pow.cpp

```diff
@@ -92,6 +92,8 @@ std::string Pow::__str__() const
     std::ostringstream o;
     if (is_a<Add>(*base_)) {
         o << "(" << *base_ << ")";
+    } else if (is_a<Mul>(*base_)) {
+        o << "(" << *base_ << ")";
```

Comment:
@thilinarmtb I can't think of any sub-condition now. I assumed that we might have them in future. 
I'm making appropriate changes and adding the case for pow.